src/3rd_party_config.c

```diff
@@ -0,0 +1,296 @@
+/*
+ *   Software Updater - client side
+ *
+ *      Copyright © 2019 Intel Corporation.
+ *
+ *   This program is free software: you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation, version 2 or later of the License.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "3rd_party_internal.h"
+#include "config.h"
+#include "lib/config_parser.h"
+#include "lib/list.h"
+#include "lib/sys.h"
+#include "swupd.h"
+#include <errno.h>
+#include <stdlib.h>
+
+#ifdef THIRDPARTY
+
+#define REPO_CONFIG_DIR "3rd_party"
+#define REPO_CONFIG_CONFIG_FILE "repo.ini"
+#define MAX_REPO_ALLOWED_FIELDS 2
+#define MAX_GLOBAL_ALLOWED_FIELDS 2
+#define MAX_FILE_PATH 1000
+
+/** @brief Storing global key,value pairs. */
+struct global {
+	void *value1;
+	void *value2;
+};
```

Comment:
I added this based on an very eatly discussion  that the repo.config file might need some global values which is applicable for whole file. 

I didn't really find the use-case for it though. I kept it all this while so that I get some feedback to keep it or remove it and add it as needed later.

If you feel its unnecessary i can remove it

Reply:
I removed it